### DISCUSSION TITLE
Feature request:  compress images from file input on all devices, for fast upload to back-end

### DISCUSSION THREAD

**Discussion by Aarbel** [ionic-team/capacitor#4389] [Ideas]
# Problem

Images coming from files inputs are basically not compressed.
These images can be very heavy, so it can be very long to upload with slow connexion (and even with a fast one).

# Feature request

Have an api / simple method to compress images coming from files inputs on the front-end side before sending them to back-end.
It should be great to use native CPU on mobile, and not webview CPU which can be limited.
Having the same returned type of files as Camera would also be great ! (‘Base64’, ‘DataUrl’ or ‘Uri’... and not Filelist).

(Images coming from Camera can be compressed and reduced in size thanks to Camera api, no problem on this side)

Thanks a lot for your help !

**Comment by Aarbel** [comment]
Browser image resizer could be a good solution, but faced this issue: https://github.com/ericnograles/browser-image-resizer/issues/39

Wkwebview doesn't seem to support classic browser APIs to reduce files images. Could be great to have a polyfill or adaptative method to compress image file.

**Comment by Aarbel** [comment]
https://github.com/autobotsrocks/cordova-imageResize or https://github.com/JoschkaSchulz/cordova-plugin-image-resizer seems to be a good start

**Reply by notflip** [reply]
These plugins are both very outdated

**Comment by sandstrom** [comment]
This is another good resize library: https://github.com/nodeca/pica

**Reply by sandstrom** [reply]
@notflip Try this one, we're using it and it works well.

**Reply by notflip** [reply]
What is the difference between using Pica or just lowering the quality and size in the CameraOptions? https://capacitorjs.com/docs/apis/camera#imageoptions

**Reply by sandstrom** [reply]
Don't know, haven't tried, but probably no difference. We're using Pica because we aren't using the Capacitor Camera Plugin to take the photos (we're using the native file element with the capture property).

**Reply by notflip** [reply]
Aha I see! Good point, is there any reason why you're not using the camera plugin?

**Reply by sandstrom** [reply]
For iOS at least, it worked just as well with the input element, and having fewer native addons makes it easier to develop in the browser, or share code with a desktop client.

We still use a bunch of native plugins, but only where the browser APIs doesn't suffice.
